src/LsCommand.cc

```diff
@@ -0,0 +1,186 @@
+/* -*- Mode: C++; tab-width: 8; c-basic-offset: 2; indent-tabs-mode: nil; -*- */
+
+#include <algorithm>
+#include <assert.h>
+#include <vector>
+
+#include "Command.h"
+#include "dirent.h"
+#include "main.h"
+#include "TraceStream.h"
+#include "util.h"
+
+using namespace std;
+
+class LsCommand : public Command {
+public:
+  virtual int run(vector<string>& args);
+
+protected:
+  LsCommand(const char* name, const char* help) : Command(name, help) {}
+
+  static LsCommand singleton;
+};
+
+LsCommand LsCommand::singleton(
+    "ls", " rr ls [OPTION]...\n"
+          "  -l, --long-listing use a long listing format \n "
+          "     (trace name | command line | start time | duration | size)\n"
+          "  -t, --sort-by-age, sort from newest to oldest\n"
+          "  -r, --reverse, the sort order\n");
+
+struct LsFlags {
+  bool reverse;
+  bool full_listing;
+  bool sort_by_time;
+  LsFlags() : reverse(false), full_listing(false), sort_by_time(false) {}
+};
+
+static bool parse_ls_arg(vector<string>& args, LsFlags& flags) {
+  if (parse_global_option(args)) {
+    return true;
+  }
+
+  static const OptionSpec options[] = { { 'r', "reverse", NO_PARAMETER },
+                                        { 'l', "long-listing", NO_PARAMETER },
+                                        { 't', "sort-by-age", NO_PARAMETER } };
+  ParsedOption opt;
+  if (!Command::parse_option(args, options, &opt)) {
+    return false;
+  }
+
+  switch (opt.short_name) {
+    case 'r':
+      flags.reverse = true;
+      break;
+    case 'l':
+      flags.full_listing = true;
+      break;
+    case 't':
+      flags.sort_by_time = true;
+      break;
+    default:
+      assert(0 && "Unknown option");
+  }
+
+  return true;
+}
+
+typedef pair<dirent*, TraceReader*> trace_info;
+
+bool compare_by_name(trace_info at, trace_info bt) {
+  auto a = string(at.first->d_name);
+  auto b = string(bt.first->d_name);
+  return lexicographical_compare(begin(a), end(a), begin(b), end(b));
+}
+
+bool compare_by_time(trace_info at, trace_info bt) {
+  auto a_version = at.second->dir() + "/version";
+  auto b_version = bt.second->dir() + "/version";
+  struct stat a_stat;
+  struct stat b_stat;
+  stat(a_version.c_str(), &a_stat);
+  stat(b_version.c_str(), &b_stat);
+  return a_stat.st_ctime < b_stat.st_ctime;
+}
+
+// http://stackoverflow.com/questions/15495756/how-to-find-the-size-of-all-files-located-inside-a-folder
+// Blatantly copied from above, slightly modified
+string get_folder_size(string path) {
+  // command to be executed
+  string cmd("du -sh " + path + " | cut -f1 2>&1 | tr -d '\\n'");
```

Comment:
Rather than depend on these utilities I'd much prefer code that just `stat`s each file in the directory and adds up the sizes. Shouldn't be much code I think.